### PR TITLE
New Documentation Component

### PR DESCRIPTION
**Pull request by kagarmoe**
The future is here!

This PR adds `components/habitat-chef-io` for our docs folder, as well as the new chef theme, config, Makefile, README, stub files, and a few other things.

To preview:
Check out this branch
Follow the steps in the README

We haven't yet settled on our pipeline for pulling the docs into the main docs.chef.io site, so some of this will change in the future. However, this gives me a stable environment for docs development.

### CODE REVIEW

**Comment by mwrock** [comment]
On windows, `make` is not something that will work natively. However any modern windows dev environment now includes the WSL (windows system for linux). Foe wxample I use the wsl for serving the middleman site. It would be great if the readme included instructions for setting up a linux environment. I tried `sudo apt-get install hugo` but it was not able to find hugo.

**Comment by kagarmoe** [comment]
Documentation added in #7551

**Comment by kagarmoe** [comment]
@mwrock I've added Linux & Windows installation instructions on #7551 (c1ede5e)

File: components/habitat-chef-io/README.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+worry about passing all of the CI checks, but it might add an extra
+few days. The important part is submitting your change.
+
+## Local Development Environment
+
+The Chef Documentation website is built using [Hugo](https://gohugo.io/) and
+[NPM](https://www.npmjs.com/). You will need Hugo 0.61 or higher installed and
+running to build and view our documentation properly.
+
+To install Hugo:
```
Comment (kagarmoe) [inline components/habitat-chef-io/README.md:42]: Windows needs a separate install
- Install Node, will install chocolatey & tools

**Comment by mwrock** [comment]
Will our website CI continue to work with this?

**Comment by kagarmoe** [comment]
@mwrock This is prep for the docs consolidation. We will continue using the existing CI until then.

**Comment by mwrock** [comment]
Probably due to some environmental issue I am not able to test the `www` site. I noticed this deletes `www/source/index.html.slim` and wanted to make sure that does not break the existing `www` site. Was that file removed intentionally?

**Comment by kagarmoe** [comment]
I've restored both the `.gitignore` and the `index.html.slim` files. We have setting up the pipeline in our plans, but it is in the queue after a few of our other products. There are also several more intermediary steps that I need to take before a complete docs build will work, such as reshaping the content and wiring up the automated parts of the documentation.

I will maintain the documentation between now and the consolidation manually because the final shape of the Hugo docs will be quite different from the current Middleman docs.

**Pull request merged**